serializations.md

```diff
@@ -229,14 +229,31 @@ subsequent sections divided by object.
 , <owner>      :: binary()
 , <vm_version> :: int()
 , <code>       :: binary()
-, <state>      :: binary(),
 , <log>        :: binary(),
 , <active>     :: bool(),
 , <referers>   :: [binary()],
 , <deposit>    :: int()
 ]
 ```
 
+The contract storage (or state) which is a key value map from (key::binary() to value::binary())
+is stored in its own subtree. The key for a contract storage value is:
+```
+<contractpubkey><16><key> :: binary()
+```
+Each value is just stored as a binary as is. If the value is the empty binary the key is pruned
+from the tree.
+
+Contracts with vm_version == 1, i.e. Sophia contracts on the AEVM stores the memory layout of the
+state as one binary value at address 0.
+
+Contracts with vm_version == 2, i.e. Solidity contracts on the AEVM
+stores the VM storage map containg 256 bit binaries for both keys and
+values in the MPT. Undefined keys are treated as having the value 0
+and keys bound to the value zero are stored as the empty binary, thus
+purging them from the tree.
```

Comment:
It looks like contract is never deleted from state tree? If it were deleted, all of this state (i.e. all keys with specific prefix) were to be deleted too, right?